client/application_msgraph.go

```diff
@@ -0,0 +1,242 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/authorization/mgmt/2018-01-01-preview/authorization"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+var _ MSGraphClient = (*AppClient)(nil)
+
+type AppClient struct {
+	Client   authorization.BaseClient
+	GraphURI string
+}
+
+type listApplicationsResponse struct {
+	Value []ApplicationResult `json:"value"`
+}
+
+func GetGraphURI(env string) (string, error) {
+	switch strings.ToUpper(env) {
+	case "AZUREPUBLICCLOUD", "":
+		return "https://graph.microsoft.com", nil
+	case "AZUREUSGOVERNMENTCLOUD":
+		return "https://graph.microsoft.us", nil
+	case "AZUREGERMANCLOUD":
```

Comment:
I'm thinking we should keep the cloud envs consistent between `GetGraphURI` and `ConfigurationFromName` wrt AzureGermanCloud. I remember it being shut down from https://github.com/hashicorp/vault-plugin-auth-azure/pull/72#discussion_r1043869392, so we should remove it here.

Reply:
Updated in latest commits